### PR TITLE
Implement `cfg80211_ops->dump_station`

### PR DESCRIPTION
**Pull request by willwillhi1**
The dump_station function is used to get statistics information for known stations. I also add validation tests to ensure proper implementation.

### CODE REVIEW

File: vwifi.c [hunk 1]
```diff
@@ -750,6 +750,63 @@ static int owl_get_station(struct wiphy *wiphy,
...
+
+    list_for_each_entry (sta_vif, &ap_vif->bss_list, bss_list) {
+        if (i < idx) {
+            ++i;
+            continue;
+        }
+        break;
+    }
+
+    if (sta_vif != ap_vif)
```
Comment (jserv) [inline vwifi.c:776]: Rewrite as following:
```c
if (sta_vif == ap_vif)
    return ret;

ret = 0;
```

**Comment by willwillhi1** [comment]
Ok, I'm going to rewrite `owl_dump_station()` to use `owl_get_station()`

**Review by jserv — CHANGES_REQUESTED** [review]
Drop the commit "Correct behavior of owl_get_station()" within this pull request.

**Comment by willwillhi1** [comment]
Because owl_dump_station() will use owl_get_station(), so I think to correct owl_get_station() before implementing owl_dump_station().

**Comment by jserv** [comment]
> Because owl_dump_station() will use owl_get_station(), so I think to correct owl_get_station() before implementing owl_dump_station().

Block this pull request until #37 is merged.

**Review by jserv — CHANGES_REQUESTED** [review]
Rebase the latest 'master' branch.

**Review by rickywu0421 — APPROVED** [review]

**Comment by jserv** [comment]
Thank @willwillhi1 for contributing!

**Pull request merged**